lib/riak.ex

```diff
@@ -306,8 +306,24 @@ defmodule Riak do
     end
   end
 
-  defp build_sibling_list([{_md, val}|t], final_list), do: build_sibling_list(t,[val|final_list])
+  # [["X-Riak-Deleted" | true]]
+  defp sibling_deleted?(md) do
+    md
+    |> Tuple.to_list
```

Comment:
This navigation of the object is concerning, I don't think it would deal well with unexpected structures for the md object. The metadata type is an Erlang dict() (https://github.com/basho/riak-erlang-client/blob/develop/src/riakc_obj.erl#L103) So perhaps we could use some of those functions to interact rather than converting tuples to lists.